### PR TITLE
Better changes detector

### PR DESCRIPTION
**Pull request by jacek-marchwicki**
Now changes detector detect order changes

### CODE REVIEW

**Comment by jacek-marchwicki** [comment]
@krpiotrek bump

File: rx-extensions/src/main/java/com/appunite/detector/ChangesDetector.java [hunk 1]
```diff
@@ -90,68 +93,69 @@ public void newData(@Nonnull ChangesAdapter adapter,
...
+            adapter.notifyItemRangeRemoved(successPosition, objects.size());
+            objects.clear();
+        }
 
-        int itemsInserted = values.size() - secondListPosition;
-        notify(adapter, counter, toRemove, itemsInserted);
         mItems = list;
     }
 
+    private boolean existInList(H[] list, int start, H search) {
```
Comment (krpiotrek) [inline rx-extensions/src/main/java/com/appunite/detector/ChangesDetector.java:139]: i'd use existing indexOfItem() method and map position < 0 to false

Comment (krpiotrek) [inline rx-extensions/src/main/java/com/appunite/detector/ChangesDetector.java:139]: please see 
https://github.com/jacek-marchwicki/rx-java-extensions/pull/83

Comment (jacek-marchwicki) [inline rx-extensions/src/main/java/com/appunite/detector/ChangesDetector.java:139]: yep.. but using array instead of list is faster

Comment (krpiotrek) [inline rx-extensions/src/main/java/com/appunite/detector/ChangesDetector.java:139]: :+1:

**Pull request merged**